### PR TITLE
Image delete auth

### PR DESCRIPTION
**Pull request by andrew21-mch**
This pull request makes the following changes:
- added second factor auth so images does not delete automatically when button is clicked

Testing checklist:
- [x] upload and image and try to delete, you will get a notification if you what to delete, click ok and its delete and cancel to maintain current photo

- [x] I certify that I ran my checklist

extra:  files changed
-  post-edit-create/media-edit.service.js
-  post-edit-create/media.html
-  post-edit-create/post-media.directive.js
-  services/media-edit-service.js

Upload photo
upload
Confirm Delete
confirm

Fixes ushahidi/platform/#4249
Ping @ushahidi/platform

### CODE REVIEW

File: legacy/app/data/common/post-edit-create/media-edit.service.js [hunk 1]
```diff
@@ -62,6 +62,12 @@ function (
             });
             return deferred.promise;
         },
+        confirmDelete: function(mediaId) {
```
Comment (Angamanga) [inline legacy/app/data/common/post-edit-create/media-edit.service.js:65]: There is already a confirmDelete function in the Notify-service you can use. By using that, you will get a dialog with styles that are adjusted to the UI. It is situated here: https://github.com/ushahidi/platform-client/blob/develop/legacy/app/common/notifications/notify.service.js#L287 and you can see how it is used for example here: https://github.com/ushahidi/platform-client/blob/develop/legacy/app/settings/categories/categories.controller.js#L44

**Review by Angamanga — CHANGES_REQUESTED** [review]
Excellent! This is exactly what is needed! However, you could make use of the already present function for confirmDelete, see my comment in the code 👍

File: legacy/app/data/common/post-edit-create/media-edit.service.js (hunk 1 above)
Comment (andrew21-mch) [reply legacy/app/data/common/post-edit-create/media-edit.service.js:65]: alright thanks, let me  check out that

**Comment by andrew21-mch** [comment]
hi @Angamanga , I just did some changes, please is this what is expexted?
update

File: legacy/app/data/common/post-edit-create/post-media.directive.js [hunk 2]
```diff
@@ -92,6 +95,12 @@ function (
                 return (!$scope.media.id && !$scope.media.changed || $scope.media.deleted);
             }
 
+            function confirmDelete(){
```
Comment (Angamanga) [inline legacy/app/data/common/post-edit-create/post-media.directive.js:98]: Remove this function

File: legacy/test/unit/mock/services/media-edit-service.js [hunk 3]
```diff
@@ -11,6 +11,9 @@ module.exports = [function () {
                     }
             };
         },
+        confirmDelete: function () {
```
Comment (Angamanga) [inline legacy/test/unit/mock/services/media-edit-service.js:14]: Same here, delete this function since its not used anymore.

**Review by Angamanga — CHANGES_REQUESTED** [review]
Thank you, this now works fine. Just remove the previous ConfirmDelete functions you did and this is good to go!

File: legacy/app/data/common/post-edit-create/post-media.directive.js (hunk 2 above)
Comment (andrew21-mch) [reply legacy/app/data/common/post-edit-create/post-media.directive.js:98]: alright, doing that right away

**Review by Angamanga — APPROVED** [review]
Approved! Thank you for your contribution 🎉

**Comment by Angamanga** [comment]
@AmTryingMyBest Ready for QA 🎉

**Comment by andrew21-mch** [comment]
thanks @Angamanga 
Am i free to continue working on other issues?

**Comment by tuxpiper** [comment]
hi @andrew21-mch ! you are more than welcome to work on other issues , let us know if we can be of assistance 🙏

**Comment by tuxpiper** [comment]
@andrew21-mch  , could you please take care of the linting problems?

```
/home/runner/work/platform-client/platform-client/legacy/app/data/common/post-edit-create/post-media.directive.js
Error:    74:14  error  Trailing spaces not allowed   no-trailing-spaces
Error:   105:38  error  Strings must use singlequote  quotes
```

you run the linting locally with `npm run lint`

**Comment by tuxpiper** [comment]
@andrew21-mch  could you please take care of the linting problems?

You can run the linter with the command `npm run lint` locally.

**Comment by andrew21-mch** [comment]
Alright will check thar out

**Comment by andrew21-mch** [comment]
Hi @tuxpiper 
I believe the linting has been fixed

**Comment by tuxpiper** [comment]
Great, thank you @andrew21-mch  !

**Comment by tuxpiper** [comment]
@AmTryingMyBest this is good to go for QA 🙏

**Review by tuxpiper — APPROVED** [review]

**Comment by tuxpiper** [comment]
Thank you @AmTryingMyBest  :+1: !

**Pull request merged**